File: frontend/src/lib/types/experimental.ts
// lib/types/experimental.ts
export interface ReferenceHypothesisBase {
    espece: string;
    race?: string;
    parametre: string;
    valeur_estimee: number;
    unite?: string;
}

export interface ReferenceHypothesisCreate extends ReferenceHypothesisBase { }

export interface ReferenceHypothesisResponse extends ReferenceHypothesisBase {
    id: number;
    utilisateur_id?: number;
    date_creation: string;
    validee: boolean;
    date_validation?: string;
    created_at: string;
}

export interface ExperimentalModeResponse {
    mode: string;
    jours_collecte: number;
    nombre_donnees_par_espece: Record<string, number>;
    confiance_moyenne: number;
    seuils_atteints: string[];
    recommandations: string[];
}

export interface ConfidenceResponse {
    prediction_type: string;
    confidence: number;
    confidence_label: string;
    facteurs: string[];
    donnees_manquantes: string[];
    recommandations: string[];
}

export interface ReferenceGenerationRequest {
    espece: string;
    force_regenerate: boolean;
}

export interface ReferenceGenerationResponse {
    espece: string;
    nouvelles_references: Array<Record<string, any>>;
    nombre_donnees_utilisees: number;
    confiance_atteinte: number;
    message: string;
}

File: backend/app/schemas/experimental.py
# backend/app/schemas/experimental.py
from pydantic import BaseModel, Field
from typing import Optional, List, Dict
from datetime import date, datetime


class ReferenceHypothesisBase(BaseModel):
    espece: str = Field(..., min_length=1, max_length=50)
    race: Optional[str] = Field(None, max_length=100)
    parametre: str = Field(..., min_length=1, max_length=100)
    valeur_estimee: float
    unite: Optional[str] = None


class ReferenceHypothesisCreate(ReferenceHypothesisBase):
    pass


class ReferenceHypothesisResponse(ReferenceHypothesisBase):
    id: int
    utilisateur_id: Optional[int]
    date_creation: date
    validee: bool
    date_validation: Optional[date]
    created_at: datetime
    
    class Config:
        from_attributes = True


class ExperimentalModeResponse(BaseModel):
    mode: str  # complet, hybride, experimental
    jours_collecte: int
    nombre_donnees_par_espece: Dict[str, int]
    confiance_moyenne: float
    seuils_atteints: List[str]
    recommandations: List[str]


class ConfidenceResponse(BaseModel):
    prediction_type: str
    confidence: float
    confidence_label: str  # Faible, Moyenne, Élevée
    facteurs: List[str]
    donnees_manquantes: List[str]
    recommandations: List[str]


class ReferenceGenerationRequest(BaseModel):
    espece: str
    force_regenerate: bool = False


class ReferenceGenerationResponse(BaseModel):
    espece: str
    nouvelles_references: List[Dict]
    nombre_donnees_utilisees: int
    confiance_atteinte: float
    message: str